Source/cursor.cpp

```diff
@@ -374,18 +374,27 @@ bool TrySelectPixelBased(Point tile)
 }
 
 #ifndef UNPACKED_MPQS
-std::vector<uint16_t> ReadWidths(const char *path)
+std::vector<uint16_t> ReadWidths(AssetRef &&ref)
 {
-	size_t len;
-	const std::unique_ptr<char[]> data = LoadFileInMem<char>(path, &len);
+	const size_t len = ref.size();
+	if (len == 0) {
+		app_fatal("Missing widths");
+	}
+	std::unique_ptr<char[]> data { new char[len] };
+
+	AssetHandle handle = OpenAsset(std::move(ref));
+	if (!handle.read(data.get(), len)) {

```

Comment:
Before attempting to `read`, the handle needs to be checked:

```suggestion
	if (!handle.ok() || !handle.read(data.get(), len)) {
```